src/serac/physics/solid.cpp

```diff
@@ -397,13 +398,82 @@ void Solid::advanceTimestep(double& dt)
   mesh_.NewNodes(*deformed_nodes_);
 
   cycle_ += 1;
+
+  previous_solve_ = PreviousSolve::Forward;
+}
+
+mfem::ParLinearForm& Solid::shearModulusSensitivity(mfem::ParFiniteElementSpace& shear_space)
+{
+  SLIC_ERROR_ROOT_IF(previous_solve_ == PreviousSolve::None,
```

Comment:
Could be worth refactoring the repeated error/warning logging code re: `previous_solve_` into an internal helper method/function?